memfs/memory.go

```diff
@@ -69,21 +75,40 @@ func (fs *Memory) OpenFile(filename string, flag int, perm os.FileMode) (billy.F
 	return f.Duplicate(filename, perm, flag), nil
 }
 
+func (fs *Memory) resolvePath(path string) string {
+	fullpath := clean(fs.Join(fs.base, path))
+	l, ok := fs.links[fullpath]
+	if !ok {
+		return fullpath
+	}
+
+	// IsAbs requires an unit so we check to if the path stars with a os.PathSeparator
+	if filepath.IsAbs(l.Target) || strings.HasPrefix(l.Target, string(os.PathSeparator)) {
+		return l.Target
+	}
+
+	return fs.resolvePath(fs.Join(filepath.Dir(fullpath), l.Target))
+}
+
 // Stat returns a billy.FileInfo with the information of the requested file.
 func (fs *Memory) Stat(filename string) (billy.FileInfo, error) {
-	fullpath := fs.Join(fs.base, filename)
-
+	fullpath := fs.resolvePath(filename)
 	f, has := fs.s.Get(fullpath)
 	if !has {
 		return nil, os.ErrNotExist
 	}
 
-	return f.Stat(), nil
+	// maybe this file belongs to a resolve link, so we use as name the
+	// filename of the opened path.
```

Comment:
I cannot understand this comment.  I suspect it is targetted to line 104 instead of 103.

Are you trying to say something along these lines:
```
The stats for a link are the stats of its target, except for its file name, which will be name of the link.
```

If this is true, I believe this comment should be public, either to the overall billy interface if that is the expected behaviour, or on the memory package.

Also, now that symlinks are supported, I think we have to explain how this functionality interacts with the rest of the API, for instance, what happens when you open, stat or close a link?  I think we should explain this in package level documentation, for the main `billy` package and for every implementation that behaves in a particularly interesting way.